src/cephadm/cephadm.py

```diff
@@ -3234,58 +3235,68 @@ def install_base_units(ctx, fsid):
 """ % (fsid, ' '.join(targets), '|'.join(targets)))
 
 
-def get_unit_file(ctx, fsid):
-    # type: (CephadmContext, str) -> str
-    extra_args = ''
-    if isinstance(ctx.container_engine, Podman):
-        extra_args = ('ExecStartPre=-/bin/rm -f %t/%n-pid %t/%n-cid\n'
-                      'ExecStopPost=-/bin/rm -f %t/%n-pid %t/%n-cid\n'
-                      'Type=forking\n'
-                      'PIDFile=%t/%n-pid\n')
-        if ctx.container_engine.version >= CGROUPS_SPLIT_PODMAN_VERSION:
-            extra_args += 'Delegate=yes\n'
-
-    docker = isinstance(ctx.container_engine, Docker)
-    u = """# generated by cephadm
+_TMPL_DAEMON_UNIT = """# generated by cephadm
 [Unit]
-Description=Ceph %i for {fsid}
+Description=Ceph %i for {{fsid}}
```

Comment:
maybe now is a good time to formally move these strs into a `.j2` (or `.jinja`) template file?

Reply:
I think that's doable. I was looking at the jinj2 docs (and source) recently and the package loader appears to support zipimport (which IIUC is the basis for zipapp).

What I'll do is try this out in a an experimental branch and if all goes smoothly I'll fold that branch back into this one. If things don't work out I'll report my findings here.